andela/magma-backend
# #168188701 User signup endpoint

#### What does this PR do?

This PR implements the user signup API endpoint

#### Description of Tasks to be completed

- Setup test suite for user signup endpoint
- Install `bcrypt` to encrypt user password
- Install `cross-env` to set environment variables
- Install `babel-eslint` to lint all babel code
- Install `nodemon` to monitor changes in code
- Install `core-js` and `@babel/runtime-corejs3` to emulate a full ES2015+ environment
- Create a helper class to generate a token and encrypt the password
- Create a utility class to handle responses
- Create a service class to mediate between database and user controller
- Create a controller class to implement signup endpoint
- Add the route
- Test with postman
- Ensure all test cases are passing
- Document with Swagger

#### How should this be manually tested?

- Clone the Repo, change into the directory and run `npm install`
- Run `npm test` to see all test cases
- Run `npm dev` to start the server
- Make a `POST` request to `http://localhost:{PORT}/api/v1/users/signup` on Postman using valid credentials

Sample request body
```
{
    "email": "naimatdavid@mail.com",
    "firstname": "Ibrahim",
    "lastname": "Jibson",
    "password": "ChibuokemAlabi99"
}
```

#### Relevant pivotal tracker story

[168188701](https://www.pivotaltracker.com/story/show/168188701)


#### Screenshots

![Screenshot (128)](https://user-images.githubusercontent.com/43535158/64128726-e9faa800-cdaf-11e9-85d7-be09c09d0f75.png)
